eventmetrics/parsers/provide.go

```diff
@@ -76,58 +105,144 @@ func provideParsers() fx.Option {
 			},
 		},
 		fx.Annotated{
-			Group:  "parsers,flatten",
-			Target: TimeElapsedParsers,
+			Group: "parsers",
+			Target: func(cycleValidator history.CycleValidator, eventValidator validation.Validator, parserIn RebootParserIn) queue.Parser {
```

Comment:
Usually a `___In` struct is the only argument for a function.  Is there a good reason why we don't want to include `cycleValidator` and `eventValidator` in `RebootParserIn`?

Reply:
Now that I've looked at this more, I think maybe this function is doing too much. We can create the parserValidators in a separate function that requires the measures, `cycleValidator`, and `eventValidator` and then pass the `parserValidators` in `RebootParserIn`.  What do you think?

Reply:
Yeah great point! I also think that the function is doing too much but I can't figure out how to break it apart. I want to set the logger with the specific parser name and pass that into the parserValidators so that when the parserValidators log, it will be clear which parser the log came from. I'm not sure how exactly to approach the problem because the parser name is set when the RebootDurationParser is created, but the RebootDurationParser needs the parserValidators, so it's kind of a chicken or egg problem...maybe have the name string as part of the `RebootParserIn` struct and tag it with a name?